content_scripts/vimium_frontend.coffee

```diff
@@ -110,8 +123,19 @@ initializePreDomReady = ->
   settings.addEventListener("load", LinkHints.init.bind(LinkHints))
   settings.load()
 
+  # Install normal mode. This is at the bottom of both the mode stack and the handler stack, and is never
+  # deactivated.
+  new NormalMode()
```

Comment:
Seeing this code here, it's kind of strange that instantiating a NormalMode also installs it. Seems to me that handler stack should be an array, and the code which is adding a handler to the handler stack actually does this, e.g.:

@handlerStack.push(new NormalMode())